vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java

```diff
@@ -120,6 +120,16 @@ default boolean isFrozenToEnd() {
     default T setTextAlign(ColumnTextAlign textAlign) {
         getElement().setProperty("textAlign",
                 textAlign == null ? null : textAlign.getPropertyValue());
+
+        // Propagate text align to parent columns
+        if (getElement().getParent() != null) {
+            getElement().getParent().getComponent()
+                    .filter(parent -> parent instanceof AbstractColumn<?>
+                            && parent.getChildren().count() == 1)
```

Comment:
This condition is untested

Reply:
Added a test that sets text-align to a column child of a joined group.